src/System.Private.CoreLib/shared/System/IO/FileStream.Unix.cs

```diff
@@ -109,7 +109,17 @@ private void Init(FileMode mode, FileShare share, string originalPath)
             {
                 // Truncate the file now if the file mode requires it. This ensures that the file only will be truncated
                 // if opened successfully.
-                CheckFileCall(Interop.Sys.FTruncate(_fileHandle, 0));
+                if (Interop.Sys.FTruncate(_fileHandle, 0) < 0)
+                {
+                    Interop.ErrorInfo errorInfo = Interop.Sys.GetLastErrorInfo();
+                    if (errorInfo.Error != Interop.Error.EBADF && errorInfo.Error != Interop.Error.EINVAL)
```

Comment:
I wonder if we should really include the EBADF here. I wonder if there can be a case when the fd becomes invalid e.g. due to a forceful unmounting of the target filesystem between the point we open the handle and the point where we call ftruncate.

Reply:
I thought about that, but such a condition could also occur immediately after. Plus the docs suggest either could be returned in the "couldn't write" case. If there's a problem, it'll fail when trying to write subsequently, which seems fine to me.

Reply:
Ok, that makes sense then.